### PR TITLE
Update bulk training progress form to work with Get Involved step

### LINKED ISSUE #2445: Update bulk add training progress form
The bulk add training progress form should work for the Get Involved step.

Extra note/bug: submitting the form for a Welcome Session added the session twice to the selected trainees, but only once? Perhaps I double-clicked submit...

Created automatically from an Asana task.

### PR DESCRIPTION
**Pull request by elichad**
In the interest of time, I'm raising this not against feature/instructor-checkout-changes but instead against the branch used in https://github.com/carpentries/amy/pull/2431 for review. Will not merge until https://github.com/carpentries/amy/pull/2431 is merged.

Fixes #2445.

This PR will:
- add fields needed for the Get Involved step to the bulk training progress form
- make the bulk form use the same JS as the individual training progress form to dynamically display fields based on options
- expand tests of the bulk form to test adding different requirements

Screenshot showing the bottom of the all trainees page. One trainee is selected and the bulk add progress form displays the fields associated with the Get Involved requirement

### CODE REVIEW

**Review by froggleston — APPROVED** [review]
LGTM 👍

File: amy/trainings/forms.py [hunk 1]
```diff
@@ -129,8 +136,10 @@ class BulkAddTrainingProgressForm(forms.ModelForm):
         # the template where this form is used
         "requirement",
         "state",
+        "involvement_type",
         "event",
         "url",
+        "date",
```
Comment (pbanaszkiewicz) [inline amy/trainings/forms.py:142]: Is there a widget for user when they click on the "date of occurrence" field?

Comment (pbanaszkiewicz) [reply amy/trainings/forms.py:142]: If not, then at least there should be a format specified in `help_text` for the field.

**Review by pbanaszkiewicz — APPROVED** [review]
Minor UX question.

File: amy/trainings/forms.py (hunk 1 above)
Comment (elichad) [reply amy/trainings/forms.py:142]: Yes, there's a widget. I think this is used by default with DateFields. It's still possible to type in a date instead, so good idea to include the required format in the help text.

**Pull request merged**